demos/http/http_demo_mutual_tls/http_demo_mutual_tls.c

```diff
@@ -590,6 +601,32 @@ static int tlsSetup( int tcpSocket,
             }
         }
 
+        if( ( sslStatus == 1 ) &&
+            ( pAlpnProtos != NULL ) && ( alpnProtosLen > 0 ) )
+        {
+            if( IOT_CORE_PORT != 443 )
+            {
+                LogError( ( "IOT_CORE_PORT must be set to 443 to use ALPN in AWS IoT Core." ) );
+                sslStatus = -1;
+            }
+            else
+            {
+                sslStatus = SSL_set_alpn_protos( *pSslContext,
+                                                 ( unsigned char * ) pAlpnProtos,
+                                                 ( unsigned int ) alpnProtosLen );
+            }
+
+            if( sslStatus != 0 )
+            {
+                LogError( ( "SSL_set_alpn_protos failed to set ALPN protos. %s",
+                            pAlpnProtos ) );
+            }
+            else
+            {
+                sslStatus = 1;
+            }
+        }
+
```

Comment:
Let's move the networking/TLS code to a separate file.  The reader has opened a file called `http_demo_mutual_tls.c`to learn about how to use the HTTP library, and the first 75% of the file has nothing to do with HTTP.  Also, moving the networking/TLS code to a separate file makes reuse possible, since that code is not HTTP-specific.

Reply:
I believe we want to push this PR in with the networking code first, then refactor later.

Reply:
Yah, the plain is to make the changes after the PR for the common networking library is finished.